src/libraries/System.Net.NameResolution/tests/FunctionalTests/HostAliasesTest.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Net.Sockets;
+using System.Threading;
+using Microsoft.DotNet.RemoteExecutor;
+using Xunit;
+
+namespace System.Net.NameResolution.Tests
+{
+    public class HostAliasesTest : FileCleanupTestBase
+    {
+        private const string AliasesEnvironmentVariableName = "DOTNET_SYSTEM_NET_HOSTALIASES";
+        private static readonly IPAddress[] s_hostNameAddresses = Dns.GetHostAddresses(Dns.GetHostName());
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void AllHookedEntrypoints_LookupsFindExpectedData()
+        {
+            string hostsPath = CreateTestFile();
+            File.WriteAllText(hostsPath, $"""
+                # This is a sample hosts aliases file
+                #
+                rhino.acme.com    {Dns.GetHostName()}       # anything
+                something.else.internal       {IPAddress.Loopback}
```

Comment:
Is it by design that we are not parsing line like `::1	localhost ip6-localhost ip6-loopback`
(found it in `docker run --rm alpine cat /etc/hosts`)

Reply:
See https://github.com/dotnet/runtime/pull/87786#issuecomment-1598534650